Objective-C/Internal/Replicator/CBLTrustCheck.mm

```diff
@@ -132,8 +141,26 @@ - (NSURLCredential*) checkTrust: (NSError**)outError {
     
     // Evaluate trust:
     SecTrustResultType result;
-    OSStatus err = SecTrustEvaluate(_trust, &result);
-    if (err) {
+    OSStatus err;
+#if TARGET_OS_MACCATALYST
+    BOOL trusted = NO;
+    CFErrorRef cfError = nullptr;
+    if (@available(iOS 12.0, macos 10.14, *)) {
+        trusted = SecTrustEvaluateWithError(_trust, &cfError);
+        err = SecTrustGetTrustResult(_trust, &result);
+    } else {
+        CBLWarnError(Sync, @"SecTrustEvaluateWithError Catalyst version not available: Not supported by macOS < 10.14 and iOS < 12.0");
+        return nil;
+    }
+#else
+    err = SecTrustEvaluate(_trust, &result);
+#endif
+    
+    if (err
+#if TARGET_OS_MACCATALYST
+        || !trusted
```

Comment:
Is it possible to clean this up? Also if trusted is false, will there be always an err?

Reply:
As per the document, yes. Will clean it with that.